### DISCUSSION TITLE
SelectionChanged unable to find setter or adder

### DISCUSSION THREAD

**Discussion by Jerkoviz** [AvaloniaUI/Avalonia#8351] [Q&A] [answered]
Hi. I have found discussion on setter and adder for Click, but it doesnt solve my problem, so I'm trying here. 
I am trying to add events in combo box and data grid, but I get this error, 
`Unable to find suitable setter or adder for property SelectionChanged`. I'm getting this error for PointerPressed and Click. Am I missing something? I am using Avalonia 0.10.14 on Windows and .Net6.  Thank you in advance. 

`<ComboBox  Name="ComboBox"
											Width="100"
											Height="28"
											Margin="15,21,0,0"
											Items="{Binding SerialCards}"
											SelectedItem="{Binding Path=SelectedCard, Mode=TwoWay}" SelectionChanged="_Some_Func"/>`
`<DataGrid MaxHeight="300"  Name="InvoicesGrid" Items="{Binding SdcImportedInvoices}" AutoGenerateColumns="False"
									  LoadingRow="dataGrid_LoadingRows" IsReadOnly="True" PointerPressed="_OnClick">`

**Comment by timunie** [comment, marked as answer]
Show your code behind as well. The event handler must match the right notation.

**Reply by timunie** [reply]
https://docs.avaloniaui.net/docs/input/routed-events

**Reply by Jerkoviz** [reply]
Thank you for reply. LoadingRow event works, but this two fail to work.
`private void dataGrid_LoadingRows(object sender, DataGridRowEventArgs e)
        {
            var dataObject = e.Row.DataContext as SdcImportedInvoice;
            if (dataObject != null)
                e.Row.Foreground = Brushes.AntiqueWhite;
        }` 
This is my code behind for theese two.  
`private void _Some_Func(object sender, RoutedEventArgs e){  }`
`private void _OnClick(object sender, RoutedEventArgs e)
        {
            int currentindex = InvoicesGrid.SelectedIndex;
        }`

**Reply by timunie** [reply]
I think the type of EventArgs may be false. It has to match exactly the event. You can look it up at the API docs. 

Or if you have aome budget use Rider. It will create the events for you.

**Reply by timunie** [reply]
In your case PointerPressedEventArgs http://reference.avaloniaui.net/api/Avalonia.Input/InputElement/EC94E2A1

**Reply by Jerkoviz** [reply]
Thank you very much. This solved my problem.  😄
